### PR TITLE
Add metric and alert for expiring certs

### PR DESCRIPTION
**Pull request by swadeley**
HMS-10103: add metric and alert for expiring certs in patch and update content-sources SOP

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

## Summary by Sourcery

Add certificate expiry monitoring for client certificates and wire it into the metrics subsystem.

New Features:
- Expose a Prometheus gauge reporting days until expiry for monitored client certificates, currently for the Candlepin client cert.
- Introduce a background collector that periodically refreshes certificate expiry metrics based on configured certificate/key material.
- Add a PrometheusRule manifest to alert when monitored client certificates are within 30 days of expiry.

Enhancements:
- Add a shared utility to compute days until expiration from a TLS certificate chain for reuse across components.

Documentation:
- Document the new certificate expiry alerting behavior in the PrometheusRule manifest annotations.

Tests:
- Add unit tests for the certificate expiry metric updater, including handling of invalid and missing certificate configuration.
- Add unit tests for the certificate expiry utility to cover normal, nil, unparsable, and empty certificate chains.

### CODE REVIEW

**Comment by dev-vektr-ai** [comment]
Thanks for the clear metric/alert addition and the focused test coverage. I found two items worth addressing before this moves out of draft:

1. `DaysTillExpiration` silently ignores certificate parse failures and returns `0, nil` when every entry in the chain is unparsable. That can make a broken or malformed cert look like an already-expired/0-day cert instead of surfacing a configuration problem, which may create confusing alerts and hide the real error source.

2. `updateCertificateExpiryGauges()` returns early on `X509KeyPair` / `DaysTillExpiration` errors without updating or clearing the gauge. If the gauge was previously set, Prometheus can keep serving a stale last-good value indefinitely. Consider explicitly resetting/deleting the `candlepin` label (or setting a separate error metric) when the cert/key cannot be parsed so operators do not get a misleading expiry value.

Otherwise the collector wiring and the alert manifest look straightforward.

File: base/metrics/metrics.go [hunk 1]
```diff
@@ -39,4 +43,7 @@ func Configure() {
 		mqueue.SetKafkaErrorReadCnt(KafkaConnectionErrorCnt.WithLabelValues("read"))
 		mqueue.SetKafkaErrorWriteCnt(KafkaConnectionErrorCnt.WithLabelValues("write"))
 	}
+	startCertificateExpiryCollector.Do(func() {
+		go runCertificateExpiryCollector(base.Context)
+	})
```
Comment (MichaelMraka) [inline base/metrics/metrics.go]: This is not a good place to check this metrics. `Configure()` runs in setup of every pod and I think we don't everyone to report expiration just of them will be enough :) . IMHO best place is `vmaas_sync`:  https://github.com/RedHatInsights/patchman-engine/blob/master/tasks/vmaas_sync/metrics.go#L115.
Also there's no need to run it in a loop via go routine.

**Review by MichaelMraka — CHANGES_REQUESTED** [review]

File: base/metrics/metrics.go (hunk 1 above)
Comment (swadeley) [reply base/metrics/metrics.go]: Hi @MichaelMraka , thank you for the review. Changes made.

**Review by MichaelMraka — APPROVED** [review]
